include/xlnt/workbook/workbook.hpp

```diff
@@ -803,6 +803,12 @@ class XLNT_API workbook
     /// </summary>
     xlnt::format create_format(bool default_format = false);
 
+    /// <summary>
+    /// Returns true if the given format belongs to this workbook's stylesheet.
+    /// Used to detect cross-workbook format references that could cause dangling pointers.
+    /// </summary>
+    bool owns_format(const class format &fmt) const;
```

Comment:
Is this something that should be part of our public interface or is this rather an implementation detail?

Reply:
I was thinking it could be useful to the users, but I am thinking better and maybe we should make it private with friendship and only expose as public interface later if really needed.